File: sniper_framework/api/routes/market.py
class SymbolsResponse(BaseModel):
    symbols: List[str]


class Candle(BaseModel):
    datetime: str
    open: float
    high: float
    low: float
    close: float
    volume: float
    ema_fast: float = 0
    ema_slow: float = 0
    ema_trend: float = 0
    rsi: float = 0
    signal: int = 0
    raw_signal: int = 0
    regime: str = "ranging"
    ml_score: float = 0
    shock_detected: bool = False
    stop_loss: Optional[float] = None
    take_profit: Optional[float] = None


class MarketDataResponse(BaseModel):
    symbol: str
    timeframe: str
    bars: int
    candles: List[Candle]
    current_regime: str
    last_updated: str


class BacktestResponse(BaseModel):
    total_trades: int
    accuracy: float
    sharpe_ratio: float
    max_drawdown: float
    profit_factor: float
    dsr_confidence: float
    total_pnl: float
    total_costs: float
    verdict: str
    accuracy_by_regime: Dict[str, float]
    cpcv: Dict[str, Any]


class RLStatusResponse(BaseModel):
    last_action: str
    confidence: float = 0
    overrode_signal: bool = False
    timestamp: Optional[str] = None


class CostSummaryResponse(BaseModel):
    entry_cost: float
    exit_cost: float
    total_cost: float
    gross_pnl: float
    net_pnl: float
    breakeven_move: float


class RiskSummaryResponse(BaseModel):
    initial_capital: float
    current_capital: float
    total_return_pct: float
    peak_capital: float
    max_drawdown_pct: float
    total_trades: int
    wins: int
    losses: int
    accuracy: float
    open_positions: int
    trading_halted: bool
    halt_reason: Optional[str] = None


class PaperOrderRequest(BaseModel):
    symbol: str
    qty: int = Field(..., ge=1)
    side: str
    order_type: str = "market"
    time_in_force: str = "day"
    market_price: float = Field(..., gt=0)
    limit_price: Optional[float] = None
    client_order_id: Optional[str] = None


class PaperOrderResponse(BaseModel):
    id: str
    symbol: str
    qty: int
    side: str
    status: str
    submitted_at: str
    filled_at: Optional[str] = None
    filled_avg_price: Optional[float] = None
    client_order_id: Optional[str] = None


class PositionResponse(BaseModel):
    symbol: str
    qty: int
    avg_entry_price: float
    updated_at: str


class HealthResponse(BaseModel):
    status: str
    timestamp: str

File: sniper-ui/types/market.ts
export type RegimeType = 'trending_bull' | 'trending_bear' | 'ranging' | 'high_vol'
export type ActionType = 'LONG' | 'SHORT' | 'HOLD' | 'NOT_LOADED'
export type DirectionType = 'LONG' | 'SHORT'

export type Candle = {
  datetime: string
  open: number
  high: number
  low: number
  close: number
  volume: number
  ema_fast: number
  ema_slow: number
  ema_trend: number
  rsi: number
  signal: number
  raw_signal: number
  regime: RegimeType
  ml_score: number
  shock_detected: boolean
  stop_loss?: number
  take_profit?: number
}

export type WSMessage =
  | { type: 'init'; data: { candles: Candle[]; symbol: string }; timestamp: string }
  | { type: 'candle_update'; data: Candle; timestamp: string }
  | { type: 'regime_change'; data: { from: RegimeType; to: RegimeType; timestamp: string }; timestamp: string }
  | { type: 'signal_fired'; data: { direction: DirectionType; ml_score: number; entry_price: number; stop_loss: number; take_profit: number; regime: RegimeType }; timestamp: string }
  | { type: 'shock_detected'; data: { vol_ratio: number; bars_cooldown: number; timestamp: string }; timestamp: string }
  | { type: 'rl_decision'; data: { action: ActionType; confidence: number; overrode_signal: boolean }; timestamp: string }
  | { type: 'alpha_warning'; data: { rolling_sharpe: number; baseline_sharpe: number; rolling_accuracy: number }; timestamp: string }
  | { type: 'risk_update'; data: { drawdown_pct: number; current_capital: number; daily_pnl_pct: number; trading_halted: boolean }; timestamp: string }
  | { type: 'error'; data: { message: string }; timestamp: string }

export type RiskSummary = {
  current_capital: number
  total_return_pct: number
  max_drawdown_pct: number
  total_trades: number
  accuracy: number
  trading_halted: boolean
  halt_reason: string | null
}

export type RLStatus = {
  last_action: ActionType
  confidence: number
  overrode_signal: boolean
  timestamp: string
}
